src/mysql_conn.erl

```diff
@@ -385,11 +385,10 @@ handle_call({prepare, Query}, _From, State) ->
             State2 = State#state{stmts = Stmts1},
             {reply, {ok, Id}, State2}
     end;
-handle_call({prepare, Name, Query}, _From, State) when is_atom(Name) ->
+handle_call({prepare, Name, Query}, _From, State) ->
     {Reply, State1} = named_prepare(Name, Query, State),
     {reply, Reply, State1};
-handle_call({unprepare, Stmt}, _From, State) when is_atom(Stmt);
-                                                  is_integer(Stmt) ->
+handle_call({unprepare, Stmt}, _From, State) ->
     case dict:find(Stmt, State#state.stmts) of
```

Comment:
I don't remember this, but do we use an integer for unnamed prepared statements, i.e. query with parameters that implicitly become a prepared statement? What happens if a user uses an integer as the name of a prepared statement? Is there a risk of collission?

If yes, then maybe we should at least forbid integers as the name, like `when not is_integer(Name)`?

Reply:
I can add and extra `is_binary(Stmt)` to be on the safe side.

Reply:
ah, you meant not to allow user explicitly unprepare using an integer.
that's new to me, I don't know why this clause was added here in the first place.